### PR TITLE
some changes

### PR DESCRIPTION
**Pull request by mikeriley131**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -8,7 +7,7 @@ function App() {
   return (
     <Layout>
       <h2>Products List</h2>
-      <ul>
+      <ul className="product-list" aria-label="Product List">
         {productData.map((item) => (
```
Comment (nmereus-turner) [inline src/App.tsx:11]: I know this was already here, and totally not part of your PR - but when I ran the project I noticed a key error in the console. We could definitely add that here as an easy win! 

```javascript
{products.map((p) => (
  <ProductCard key={p.id} productData={item} />
))}
```

File: src/App.tsx [hunk 2]
```diff
@@ -8,7 +7,7 @@ function App() {
...
           <ProductCard productData={item} />
```
Comment (graphicallychallenged) [inline src/App.tsx:12]: I know this was already here, and totally not part of your PR - but when I ran the project I noticed a key error in the console. We could definitely add that here as an easy win!

```javascript
{products.map((p) => (
  <ProductCard key={p.id} productData={item} />
))}
```

File: src/styles/main.scss [hunk 3]
```diff
@@ -11,5 +11,7 @@
 @import 'partials/components/header';
 @import 'partials/components/footer';
 @import 'partials/components/intro-card';
+@import 'partials/components/product-list';
+@import 'partials/components/product-card';
 //@import 'partials/html-linter'; // for development only
 @import 'partials/mq-helper'; // for development only
```
Comment (graphicallychallenged) [inline src/styles/main.scss:17]: should this be commented out if its development only?

File: src/components/ProductCard.tsx [hunk 4]
```diff
@@ -1,16 +1,31 @@
-export default function ProductCard({ productData }) {
+import React, { useState } from 'react';
+import _ from 'lodash';
+
+type Props = {
+  productData: any;
```
Comment (graphicallychallenged) [inline src/components/ProductCard.tsx:5]: Looks like we do have some idea of the shape of the data, based on the productData file, we probably should avoid `any` and try these types: 

```javascript
type Props = {
  id: number
  imgSrc: string
  name: string
  author: string
  description: string
  features: string[]
  price: number
}
``` 

For a cleaner experience we should really put this in the productData.ts and then import it, but more of a nitpik than a requirement :-)

File: src/components/ProductCard.tsx [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
-        <li>{productData.imgSrc}</li>
-        <li>{productData.name}</li>
-        <li>{productData.author}</li>
-        <li>{productData.description}</li>
-        <li>add to cart button</li>
-        <li>like button</li>
-        <li>- full or empty heart emoji</li>
-        <li>product features list</li>
+    <div className="product-card" onClick={handleClick}>
+      <img src={productData.imgSrc} />
```
Comment (graphicallychallenged) [inline src/components/ProductCard.tsx:18]: For accessibility, we should add an "alt" tag here with the product name

File: src/components/ProductCard.tsx [hunk 6]
```diff
@@ -1,16 +1,31 @@
-export default function ProductCard({ productData }) {
+import React, { useState } from 'react';
+import _ from 'lodash';
```
Comment (graphicallychallenged) [inline src/components/ProductCard.tsx:2]: This doesn't seem to be used anywhere in the file, is it needed? (I also got an error locally that it wasn't installed as part of the project - we should confirm this is added to package file if we want to use it, jic)

File: src/components/ProductCard.tsx [hunk 7]
```diff
@@ -1,16 +1,31 @@
...
-        <li>- full or empty heart emoji</li>
-        <li>product features list</li>
+    <div className="product-card" onClick={handleClick}>
+      <img src={productData.imgSrc} />
+      <h2>{productData.name}</h2>
+      <h5>{productData.author}</h5>
+      <p>{productData.description}</p>
+      <ul className="aligned-list">
+        {productData.features.map((f: any, i: number) => (
+          <li key={i}>{f}</li>
```
Comment (graphicallychallenged) [inline src/components/ProductCard.tsx:24]: We should use a unique identifier here if possible, other than the index

File: src/components/ProductCard.tsx [hunk 8]
```diff
@@ -1,16 +1,31 @@
...
+        ))}
       </ul>
+      <button onClick={() => alert('Added to cart!')}>Add</button>
```
Comment (graphicallychallenged) [inline src/components/ProductCard.tsx:27]: For accessibility, we may want to make this button a little more descriptive if possible. We can check with design and see if they are ok with "Add to cart"

File: src/styles/partials/components/_product-card.scss [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  button {
+    background: blue;
+    color: white;
+    padding: 5px 10px;
+    border: none;
+    cursor: pointer;
+
+    &:hover {
+      background: darkblue !important;
+    }
```
Comment (graphicallychallenged) [inline src/styles/partials/components/_product-card.scss:38]: Did the designs include a focus state? We should try to implement if so, and if not discuss with design team

File: src/styles/partials/components/_product-card.scss [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+  button {
+    background: blue;
+    color: white;
+    padding: 5px 10px;
+    border: none;
+    cursor: pointer;
+
+    &:hover {
+      background: darkblue !important;
```
Comment (graphicallychallenged) [inline src/styles/partials/components/_product-card.scss:37]: I don't think you have to overwrite this with !important -  by simply being in this order, your darkblue will win  :-)  - also, can we create or use a token here?

File: src/styles/partials/components/_product-card.scss [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+  height: 600px;
+  background-color: $color-grey-8;
+  color: #999;
+  font-family: Arial;
+  border: 1px solid #ccc;
+  padding: 1rem;
+  border-radius: 8px;
+
+  img {
+    float: left;
```
Comment (graphicallychallenged) [inline src/styles/partials/components/_product-card.scss:13]: nitpick: float can be super problematic - Can we try a flex solution here instead?

File: src/tests/ProductCard.test.tsx [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+          name: 'Cart Product',
+          author: 'Some Author',
+          description: 'Cart Desc',
+          imgSrc: '/cart.png',
+          features: [],
+        }}
+      />,
+    );
+
+    fireEvent.click(screen.getByText('Add'));
```
Comment (graphicallychallenged) [inline src/tests/ProductCard.test.tsx:54]: We need to also assert some outcomes and confirm them here - currently we are just clicking. Instead let's add some assertions about alerts or ui changes to confirm all of the appropriate actions are being taken.

You can also test for accessible roles, ie. 
`getByRole('button', { name: /add/i })`

File: src/tests/ProductCard.test.tsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+      name: 'Feature Product',
+      author: 'Some Author',
+      description: 'Desc',
+      imgSrc: '/feature.png',
+      features: ['feature1', 'feature2', 'feature3'],
+    };
+
+    render(<ProductCard productData={product} />);
+
+    expect(screen.getByText('feature2')).toBeTruthy();
```
Comment (graphicallychallenged) [inline src/tests/ProductCard.test.tsx:68]: I would check for three features here, and not just one, based on the test name

File: src/tests/ProductCard.test.tsx [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+          name: 'Test Product',
+          description: 'This is a test product',
+          author: 'Some Author',
+          imgSrc: '/test.png',
+          features: ['one', 'two', 'three'],
+        }}
+      />,
+    );
+
+    expect(screen.getByText('Test Product')).toBeTruthy();
```
Comment (graphicallychallenged) [inline src/tests/ProductCard.test.tsx:19]: We can also check here for img src being correct and  alt tags being in place

File: src/data/productData.ts [hunk 15]
```diff
@@ -77,7 +77,20 @@ export default [
     features: [
       'Fresh perspective to reexamine the principles',
       'Updated examples and a new chapter on mobile usability',
-      'feature 3',
```
Comment (graphicallychallenged) [inline src/data/productData.ts:80]: Seems like in the tests, it was asserted that each card should have three features - should this be replaced with a relevant feature instead of removed entirely?

File: src/styles/partials/components/_product-card.scss [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    font-weight: bold;
+  }
+
+  p {
+    font-size: 14px;
+    color: #666;
+  }
+
+  button {
+    background: blue;
```
Comment (graphicallychallenged) [inline src/styles/partials/components/_product-card.scss:30]: Is there a token for this value that we could leverage?

**Review by graphicallychallenged — COMMENTED** [review]
Overall, looks fairly good with some minor tweaks needed. Some feedback had questions to help me understand just in case my suggestion may not be relevant :-) 

For the tests, I definitely would suggest adding some edge cases for errors, missing info, etc, and for the others just add more assertions to check that the data is rendering as expected. Let me know if you have any questions, and I can give it another look to address any formatting, etc once we have gotten past this round of quick fixes! 

Thanks so much :-)
